### ISSUE TITLE
Unhandled rejection TypeError: Cannot read property 'columns' of undefined

### ISSUE THREAD

**Issue by anodynos** [Daplie/serve-https#4]
Tried with node@6.9, @6.3, @4.4.2 on Kubuntu 16.04 x64 with the same results: 
```
╰─$ serve-https --agree-tos --email agelos.pikoulas@gmail.com
Unhandled rejection TypeError: Cannot read property 'columns' of undefined
    at Object.FNS.reComputeHelper (/devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/cli.js:407:16)
    at Object.FNS.reCompute (/devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/cli.js:166:13)
    at handleInput (/devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/cli.js:196:7)
    at /devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/cli.js:1226:5
    at Promise._execute (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/debuggability.js:299:9)
    at Promise._resolveFromExecutor (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:77:14)
    at Object.CLI.readProviderUrlAsync (/devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/cli.js:1225:10)
    at Object.Oauth3._restoreConfig (/devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/index.js:624:22)
    at loopit (/devtools/npm-global/lib/node_modules/serve-https/node_modules/oauth3-cli/lib/index.js:613:19)
    at tryCatcher (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:636:18)
    at Promise._resolveCallback (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:431:57)
    at Promise._settlePromiseFromHandler (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:522:17)
    at Promise._settlePromise (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/devtools/npm-global/lib/node_modules/serve-https/node_modules/bluebird/js/release/promise.js:636:18)
```

**Comment by anodynos** [comment]
Funny thing - If I use `&` at the end it doesnt throw exception  
```
╰─$ serve-https --agree-tos --email jane@example.com  &
[6] 9509
```
but then I receive a
```
[6]  + 9509 suspended (tty output)  serve-https --agree-tos --email jane@example.com
```
and it doesnt work...

**Comment by coolaj86** [comment]
It's a module version mismatch (that function doesn't do like that anymore). I'll have to take a look on my machine and make sure the right versions are published and listed in package.json

At a conference today so I may or may not have time until Friday.

**Comment by claytoncarney** [comment]
I encountered the same issue as @anodynosa on Windows 10 using the following command:

`serve-https --agree-tos --email clayton.carney@gmail.com`

**Comment by joshuazsprague** [comment]
same issue here
`
# serve-https --agree-tos --email foo@gmail.com
Unhandled rejection TypeError: Cannot read property 'columns' of undefined
`
`
# node -v
v6.9.2
`
`
# sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G31
`

**Comment by coolaj86** [comment]
Will investigate...

**Comment by coolaj86** [comment]
Okay, just downloaded the latest version and tested on OS X which works and then tested on Ubuntu to see that it does not work.

Here's part of what's happening:

Because you're using this for the first time you're being asked to validate your email address. The prompt for the validation uses a library I wrote which tries to determine the column width - because some prompts in the library place helper text or auto-hinting to the right - even though this specific prompt doesn't need the column width.

The reason you're getting the error is that the console object, for some reason, is not defined. Normally this would be due to some sort of terminal emulator (tmux, screen) or terminal (msysgit) that doesn't support reading the column-width on the operating system that you're using.

However, in this case it's actually that the console object isn't even passed in... but only on some operating systems...

Anyway, I'm working on it now.

Also, we've moved to https://git.daplie.com/Daplie/serve-https

**Comment by coolaj86** [comment]
It happens when you don't already have an account.

Workarounds:

1) If you're not trying to use Let's Encrypt on a real domain through Daplie Domains, use the older version:

```
npm install -g serve-https@1.x
```

2) If you are wanting to use Let's Encrypt with Daplie Domains, you should have created an account with `daplie-tools` already:

```
npm install -g daplie-tools

daplie login

# ouputs some debugging stuff, don't worry about it
```

**Comment by coolaj86** [comment]
Found the bug, but I have to leave right now. If any of you would like to test and pull request, it's this:

https://git.daplie.com/Daplie/serve-https/issues/5

**Comment by coolaj86** [comment]
Fixed in v2.0.7
